packages/ember/tests/routing/substates_test.js

```diff
@@ -1,1042 +1,880 @@
-import { RSVP, Controller } from 'ember-runtime';
-import { Route, NoneLocation } from 'ember-routing';
-import { run } from 'ember-metal';
-import { compile } from 'ember-template-compiler';
-import { Application, Resolver } from 'ember-application';
-import { jQuery } from 'ember-views';
-import { setTemplates, setTemplate } from 'ember-glimmer';
-
-let Router, App, templates, router, container, counter;
-
-function step(expectedValue, description) {
-  equal(counter, expectedValue, 'Step ' + expectedValue + ': ' + description);
-  counter++;
-}
-
-function bootApplication(startingURL) {
-  for (let name in templates) {
-    setTemplate(name, compile(templates[name]));
+import { RSVP } from 'ember-runtime';
+import { Route } from 'ember-routing';
+import {
+  moduleFor,
+  ApplicationTestCase,
+  AutobootApplicationTestCase } from 'internal-test-helpers';
+
+moduleFor('Loading/Error Substates', class extends ApplicationTestCase {
+  constructor() {
+    super();
+
+    this.addTemplate('application', `<div id="app">{{outlet}}</div>`);
+    this.addTemplate('index', 'INDEX');
   }
 
-  if (startingURL) {
-    NoneLocation.reopen({
-      path: startingURL
-    });
+  getController(name) {
+    return this.applicationInstance.lookup(`controller:${name}`);
   }
 
-  startingURL = startingURL || '';
-  router = container.lookup('router:main');
-  run(App, 'advanceReadiness');
-}
-
-QUnit.module('Loading/Error Substates', {
-  setup() {
-    counter = 1;
-
-    run(function() {
-      App = Application.create({
-        name: 'App',
-        rootElement: '#qunit-fixture',
-        // fake a modules resolver
-        Resolver: Resolver.extend({ moduleBasedResolver: true })
-      });
-
-      App.deferReadiness();
-
-      App.Router.reopen({
-        location: 'none'
-      });
-
-      Router = App.Router;
-
-      container = App.__container__;
+  ['@test Slow promise from a child route of application enters nested loading state'](assert) {
+    let turtleDeferred = RSVP.defer();
 
-      templates = {
-        application: '<div id="app">{{outlet}}</div>',
-        index: 'INDEX',
-        loading: 'LOADING',
-        bro: 'BRO',
-        sis: 'SIS'
-      };
+    this.router.map(function() {
+      this.route('turtle');
     });
-  },
-
-  teardown() {
-    run(function() {
-      App.destroy();
-      App = null;
-
-      setTemplates({});
-    });
-
-    NoneLocation.reopen({
-      path: ''
-    });
-  }
-});
 
-QUnit.test('Slow promise from a child route of application enters nested loading state', function() {
-  let broModel = {};
-  let broDeferred = RSVP.defer();
-
-  Router.map(function() {
-    this.route('bro');
-  });
-
-  App.ApplicationRoute = Route.extend({
-    setupController() {
-      step(2, 'ApplicationRoute#setup');
```

Comment:
Isn't some coverage lost by dropping these `step` assertions for what hooks fire in what order?